packages/contracts-bedrock/contracts/dispute/AttestationDisputeGame.sol

```diff
@@ -0,0 +1,292 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import { EIP712 } from "@solady/utils/EIP712.sol";
+import { ECDSA } from "@solady/utils/ECDSA.sol";
+
+import { Initializable } from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+
+import { Hash } from "../libraries/DisputeTypes.sol";
+import { Claim } from "../libraries/DisputeTypes.sol";
+import { GameType } from "../libraries/DisputeTypes.sol";
+import { Timestamp } from "../libraries/DisputeTypes.sol";
+import { GameStatus } from "../libraries/DisputeTypes.sol";
+
+import { InvalidSignature } from "../libraries/DisputeErrors.sol";
+import { AlreadyChallenged } from "../libraries/DisputeErrors.sol";
+
+import { IAttestationDisputeGame } from "./IAttestationDisputeGame.sol";
+import { IBondManager } from "./IBondManager.sol";
+import { Clone } from "../libraries/Clone.sol";
+import { SystemConfig } from "../L1/SystemConfig.sol";
+import { L2OutputOracle } from "../L1/L2OutputOracle.sol";
+
+/**
+ * @title AttetationDisputeGame
+ * @notice A contract for disputing the validity of a claim via permissioned attestations.
+ */
+contract AttestationDisputeGame is Initializable, IAttestationDisputeGame, Clone, EIP712 {
+    /**
+     * @notice The EIP-712 type hash for the `Dispute` struct.
+     */
+    Hash public constant DISPUTE_TYPE_HASH =
+        Hash.wrap(keccak256("Dispute(bytes32 outputRoot,uint256 l2BlockNumber)"));
+
+    /**
+     * @notice The current version of the `AttestationDisputeGame` contract.
+     */
+    string public constant VERSION = "0.0.1";
```

Comment:
I don't think we need a getter for both `version` and `VERSION`. Do you want to move away from `Semver`? Its slightly wasteful, bloating the contract size, but it does enforce standardization

Reply:
I agree. This is only to adhere to the interface, which we'd be better off updating.

Reply:
We could turn `Semver` into an abstract contract in future releases and move away from the current immutable style

Reply:
Marked internal, not going to introduce `Semver` changes to avoid a larger frozen code diff.